src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -209,9 +207,117 @@ public async Task<JsonSchemaValidationResult> ValidateConfigSchema(RuntimeConfig
     }
 
     /// <summary>
-    /// This method runs validates the entities metadata against the database objects.
+    /// Validates the semantic correctness of an Entity's relationships in the
+    /// runtime configuration when metadata from the backend DB is not required.
+    /// Validating Cases:
```

Comment:
```suggestion
    /// runtime configuration without crossreferencing database metadata.
    /// Validating Cases:
```

"not required" is ambiguous. It could imply:
- when database metadata is available it could be used
- user can supply database metadata somehow to alter how this function validates config.